src/components/input-field/partial-styles/trailing-icon.scss

```diff
@@ -1,11 +1,11 @@
 @use '../../../style/functions';
 
 .mdc-text-field--with-trailing-icon {
-    .mdc-text-field__icon.trailing-icon {
+    .mdc-text-field__icon--trailing {
         transition: background-color 0.2s ease, box-shadow 0.3s ease;
         border-radius: 50%;
         padding: functions.pxToRem(8);
-        right: functions.pxToRem(8);
+        margin-right: functions.pxToRem(8);
```

Comment:
This was fine actually to be `right` and not `margin-right`. The problem is that some CSS that should be coming from Material Design styles is missing.

The thing is that `mdc-text-field__icon` should have an absolute position, which comes from these styles:
```css
.mdc-text-field--with-leading-icon .mdc-text-field__icon, 
.mdc-text-field--with-trailing-icon .mdc-text-field__icon {
    position: absolute;
    top: 50%;
    transform: translateY(-50%);
    cursor: pointer;
}
```
![image](https://user-images.githubusercontent.com/35954987/125785205-eb9f0674-2a17-4460-8cc2-c63305c0b601.png)

But for some reason they don't appear here, which messes up the UI.

I'll revert this change in a fixup soon. But I can't really figure out why the MD styles are not coming. All class names seem to be fine :/

Reply:
Again, _please_ use the actual docs for the actual MDC version we're using as your primary guidelines. If something doesn't work the way it's supposed to when using those, then go ahead and change.

If there are styles missing, we're probably missing a `@include textfield.core-styles;` or other mixin somewhere. The way styles are included have changed completely between MDC v4 and v11.

Reply:
https://github.com/material-components/material-components-web/tree/v11.0.0/packages/mdc-textfield

Reply:
Awesome. I'll check there as well now. 

In this case, the `right: functions.pxToRem(8);` was added by us to override MD's default `right`. This is because we add some hover effects to those icons and render them like clickable buttons and so forth. Thus, we needed some certain amount of space on the right side.

However, it's hard to say, without getting those expected styles form MDC. `@include textfield.core-styles;` and `@include icon.icon-core-styles;` seem to be there in `input-field.scss`. Must we add it elsewhere? I don't know where and how 😞

Reply:
I'm not entirely sure what is missing and where… I'll ping you in Slack, and maybe you can show me.

Reply:
Ah, wait, I think this has become the same discussion as https://github.com/Lundalogik/lime-elements/pull/1297#discussion_r670450262, right? If so, let's just keep it in that thread instead 😊